runtime/doc/lua.txt

```diff
@@ -3388,6 +3388,29 @@ Iter:find({f})                                                   *Iter:find()*
     Return: ~
         any
 
+Iter:flatten({depth})                                         *Iter:flatten()*
+    Flattens a |list-iterator| pipeline, un-nesting nested values. The
+    flattened list will exclude nil, and any values with non-numeric keys
+
+    Example:
+
+    >lua
+        vim.iter({1, {2}, {{3}}}):flatten():totable()
+
+    <(returns `{1, 2, {3}}` )
+
+    >lua
+        vim.iter({1, {a = 2}, {nil, {nil}}, {3}}):flatten():totable()
+
+    <(returns `{1, {}, 3}` )
```

Comment:
Seems strange that it oblietrates nested maps. Was this already discussed? How are maps/dicts treated in other flatten() implementations (e.g. JS, erlang)?

```suggestion
    <(returns `{1, {a = 2}, 3}` )
```